cluster-autoscaler/cloudprovider/aws/aws_wrapper.go

```diff
@@ -233,32 +236,151 @@ func (m *awsWrapper) getInstanceTypeByLaunchTemplate(launchTemplate *launchTempl
+		Versions:           []*string{aws.String(policy.launchTemplate.version)},
+	}
+
+	start := time.Now()
+	describeData, err := m.DescribeLaunchTemplateVersions(describeTemplateInput)
+	observeAWSRequest("DescribeLaunchTemplateVersions", err, start)
+	if err != nil {
+		return "", err
+	}
 	if len(describeData.LaunchTemplateVersions) == 0 {
 		return "", fmt.Errorf("unable to find template versions")
 	}
 
-	lt := describeData.LaunchTemplateVersions[0]
-	instanceType := lt.LaunchTemplateData.InstanceType
+	requirements, err := m.getRequirementsRequestFromAutoscaling(policy.instanceRequirementsOverrides)
+	if err != nil {
+		return "", err
+	}
+	instanceType, err := m.getInstanceTypeFromLaunchTemplateData(*describeData.LaunchTemplateVersions[0].LaunchTemplateData.ImageId, requirements)
+	if err != nil {
+		return "", err
+	}
 
-	if instanceType == nil {
-		return "", fmt.Errorf("unable to find instance type within launch template")
+	return instanceType, nil
+}
+
+func (m *awsWrapper) getInstanceTypeFromLaunchTemplateData(imageId string, requirementsRequest *ec2.InstanceRequirementsRequest) (string, error) {
+	describeImagesInput := &ec2.DescribeImagesInput{
+		ImageIds: []*string{aws.String(imageId)},
 	}
 
-	return aws.StringValue(instanceType), nil
+	start := time.Now()
+	describeImagesOutput, err := m.DescribeImages(describeImagesInput)
+	observeAWSRequest("DescribeImages", err, start)
+	if err != nil {
+		return "", err
+	}
+
+	imageArchitectures := []*string{}
+	imageVirtualizationTypes := []*string{}
+	for _, image := range describeImagesOutput.Images {
+		imageArchitectures = append(imageArchitectures, image.Architecture)
+		imageVirtualizationTypes = append(imageVirtualizationTypes, image.VirtualizationType)
+	}
+
+	requirementsInput := &ec2.GetInstanceTypesFromInstanceRequirementsInput{
+		ArchitectureTypes:    imageArchitectures,
+		InstanceRequirements: requirementsRequest,
+		VirtualizationTypes:  imageVirtualizationTypes,
+	}
+
+	start = time.Now()
+	instanceTypes := []string{}
+	err = m.GetInstanceTypesFromInstanceRequirementsPages(requirementsInput, func(page *ec2.GetInstanceTypesFromInstanceRequirementsOutput, isLastPage bool) bool {
+		for _, instanceType := range page.InstanceTypes {
+			instanceTypes = append(instanceTypes, *instanceType.InstanceType)
+		}
+		return !isLastPage
+	})
+	observeAWSRequest("GetInstanceTypesFromInstanceRequirements", err, start)
+	if err != nil {
+		return "", fmt.Errorf("unable to get instance types from requirements")
+	}
+
+	return instanceTypes[0], nil
+}
+
+func (m *awsWrapper) getRequirementsRequestFromAutoscaling(requirements *autoscaling.InstanceRequirements) (*ec2.InstanceRequirementsRequest, error) {
+	requirementsJson, err := json.Marshal(*requirements)
+	if err != nil {
+		return nil, err
+	}
+
+	var requirementsRequest *ec2.InstanceRequirementsRequest
+	if err := json.Unmarshal(requirementsJson, &requirementsRequest); err != nil {
+		return nil, err
+	}
```

Comment:
heh. sneaky. But prone to breakage if and when the shapes in the EC2 and EC2 autoscaling APIs inevitably change... recommend rewriting this using a more verbose constructor and field setting approach instead of the JSON-to-JSON approach.